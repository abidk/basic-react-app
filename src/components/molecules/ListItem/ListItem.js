import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Hello from '../../atoms/Hello';
import Button from '../../atoms/Button';
import './ListItem.css';

class ListItem extends Component {

    constructor(props) {
        super(props);
        this.state = { isVisible: true };
        this.toggleBox = this.toggleBox.bind(this);
    }

    toggleBox() {
        this.setState(oldState => ({ isVisible: !oldState.isVisible }));
    }

    render() {
        const { isVisible } = this.state;

        return (
            <li className="ListItem">
                {isVisible && <Hello isVisible={isVisible} {...this.props} />}
                <Button label={`Show/Hide ${this.props.name}`} action={this.toggleBox} />
            </li>
        )
    }
}

ListItem.defaultProps = {
    name: "Joe Bloggs"
}

ListItem.propTypes = {
    name: PropTypes.string.isRequired
}

export default ListItem;
