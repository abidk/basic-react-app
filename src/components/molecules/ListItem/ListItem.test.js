import React from 'react'
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ListItem from './ListItem'

describe(__filename, () => {

    it('Should render', () => {
        const output = mount(<ListItem name="Abid" />);

        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should not show Hello', () => {
        const output = mount(<ListItem name="Abid" />);
        output.find('button').simulate('click');

        expect(output.state('isVisible')).toEqual(false);

        expect(shallowToJson(output)).toMatchSnapshot();
    });
});