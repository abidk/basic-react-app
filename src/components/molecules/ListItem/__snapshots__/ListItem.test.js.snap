// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/abidkhalil/Documents/basic-react-app/src/components/molecules/ListItem/ListItem.test.js Should render 1`] = `
<ListItem
  name="Abid"
>
  <li
    className="ListItem"
  >
    <Hello
      isVisible={true}
      name="Abid"
    >
      <p
        className="HelloText"
      >
        Hello 
        Abid
      </p>
    </Hello>
    <Button
      action={[Function]}
      label="Show/Hide Abid"
    >
      <button
        className="Button"
        onClick={[Function]}
      >
        Show/Hide Abid
      </button>
    </Button>
  </li>
</ListItem>
`;

exports[`/Users/abidkhalil/Documents/basic-react-app/src/components/molecules/ListItem/ListItem.test.js should not show Hello 1`] = `
<ListItem
  name="Abid"
>
  <li
    className="ListItem"
  >
    <Button
      action={[Function]}
      label="Show/Hide Abid"
    >
      <button
        className="Button"
        onClick={[Function]}
      >
        Show/Hide Abid
      </button>
    </Button>
  </li>
</ListItem>
`;
